src/Orleans/AzureUtils/AzureTableDataManager.cs

```diff
@@ -152,7 +150,7 @@ public AzureTableDataManager(string tableName, string storageConnectionString, T
         /// </summary>
         /// <param name="data">Data to be inserted into the table.</param>
         /// <returns>Value promise with new Etag for this data entry after completing this storage operation.</returns>
-        public async Task<string> CreateTableEntryAsync(T data)
+        public async Task<string> TryCreateTableEntryAsync(T data)
```

Comment:
Yes, a very good point. The method will throw if failed to create. In fact, all Update/Insert/Merge/Delete may throw, for various reasons, Conflict is only one of them. I don't think we want to change them all to be TryXXX  and handle conflict error differently from other errors.

So seems like the right thing to do is to rename this method back to Create, just like it was before and let the caller deal with all the different errors, just like it does now for all those methods.
In the specific case of Azure table membership the caller wraps the Create with try catch and returns Task<bool> upon success, thus adding its own behavior to this pattern. But that is only in this specific case.
I will update the PR.